block-layer/codecs/DAG-JSON.md

```diff
@@ -14,14 +14,56 @@ Contrary to popular belief, JSON as a format supports Big Integers. It's only
 JavaScript itself that has trouble with them. This means JS implementations
 of `dag-json` can't use the native JSON parser and serializer.
 
+### Version 0
+
+This is an old version of `dag-json` that reserved the `"/"` key in order to
+represent binary and link data types.
+
 #### Binary Type
 
 ```javascript
 {"/": { "base64": String }}
 ```
 
-### Link Type
+#### Link Type
 
 ```javascript
 {"/": String /* base encoded CID */}
-```
\ No newline at end of file
+```
+
+### Version 1
+
+#### Format
+
+The internal data format is valid JSON but is **NOT** identical to the decoded
+node data codecs produce.
+
+Example internal format:
+
+```javascript
+{ "/": { "_": 1 },
+  "data": { "hello": "world", { "obj": { "array": [0, 0] } } },
+  "meta": {
+    base64: [
+      [[ "key" ], "dmFsdWU="],
```

Comment:
I key see where those tuples come from. Why not using an "overlay object" with proper nesting as you did in your other dag-json experiment?

Is it because you could get the binary data/links without traversing a nested structure?

Reply:
If you recall, we considered that in a `dag-json` alternative proposal and the problem, which I think you ended up alerting me to, was how to represent sparse values in an array.

Reply:
Right, that would be a lot of placeholders then.